lib/iris/analysis/__init__.py

```diff
@@ -1419,36 +1419,51 @@ def interp_order(length):
 """
 
 
-def _build_dask_mdtol_function(dask_nanfunction):
+def _build_dask_mdtol_function(dask_stats_function):
     """
     Make a wrapped dask statistic function that supports the 'mdtol' keyword.
 
-    'dask_nanfunction' must be a statistical function that treats NaNs as
-    missing data, and which has the basic call signature
-    "dask_nanfunction(data, axis, **kwargs)".
+    'dask_function' must be a dask statistical function, compatible with the
+    call signature : "dask_stats_function(data, axis, **kwargs)".
+    It must be masked-data tolerant, i.e. it ignores masked input points and
+    performs a calculation on only the unmasked points.
+    For example, mean([1, --, 2]) = (1 + 2) / 2 = 1.5.
 
     The returned value is a new function operating on dask arrays.
-    It has the call signature "stat(data, axis=-1, mdtol=None, *kwargs)".
+    It has the call signature `stat(data, axis=-1, mdtol=None, **kwargs)`.
 
     """
-    @wraps(dask_nanfunction)
+    @wraps(dask_stats_function)
     def inner_stat(array, axis=-1, mdtol=None, **kwargs):
-        dask_result = dask_nanfunction(array, axis=axis, **kwargs)
+        # Call the statistic to get the basic result (missing-data tolerant).
+        dask_result = dask_stats_function(array, axis=axis, **kwargs)
         if mdtol is None:
             result = dask_result
         else:
-            point_counts = np.sum(np.ones(array.shape), axis=axis)
-            point_mask_counts = da.sum(da.isnan(array), axis=axis)
-            masked_point_fractions = (point_mask_counts + 0.0) / point_counts
-            # Note: the +0.0 forces a floating-point divide.
+            # Build a lazy computation to compare the fraction of missing
+            # input points at each output point to the 'mdtol' threshold.
+            if axis is not None:
+                # Make an iterable over one or multiple axis values.
+                axis_indices = np.array(axis, ndmin=1)
+            else:
+                # Collapse is over all axes.
+                axis_indices = range(array.ndim)
+            # Multiply the sizes of the collapsed dimensions, to get
+            # the total number of input points at each output point.
+            point_counts = np.prod([array.shape[axis_index]
+                                    for axis_index in axis_indices])
```

Comment:
How about:
```
point_counts = np.prod(array.shape) / np.prod(dask_result.shape)
```

Reply:
Dead right !! 👍 
But I get even more ...
as "np.prod(a.shape)" === "a.size", we can just use that...
See following commit,